### ISSUE TITLE
Compatibility with OpenMPI

### ISSUE THREAD

**Issue by jayeshkrishna** [E3SM-Project/scorpio#295]
Ensure that Scorpio builds and Scorpio tests run successfully with OpenMPI.

At least with OpenMPI 4.0.2/4.0.3,

* OpenMPI needs to be built with "--enable-mpi1-compatibility" option. Without this option PnetCDF 1.12.0 build fails. OpenMPI default builds no longer offer MPI 1.0 compatibility
* On anlworkstation (compute001) a majority of the Scorpio tests fail with OpenMPI 4.0.2/4.0.3. Note that these tests pass with MPICH.

**Comment by jayeshkrishna** [comment]
I have OpenMPI 4.0.2/4.0.3 installed on anlworkstation at /soft/apps/packages/climate/openmpi/* and PnetCDF libraries built with OpenMPI at /soft/apps/packages/climate/pnetcdf/1.12.0/openmpi-4.0.* .

**Comment by jayeshkrishna** [comment]
@dqwu was able to run the tests successfully with the OpenMPI installation on anlworkstation. After debugging the test setup we finally found that the failures in my runs were caused due to using the default mpiexec (instead of the OpenMPI's version of mpiexec).

Adding OpenMPI installation path (bin) to PATH and rerunning the tests I was able to run the tests successfully. @dqwu also verified that using the default mpiexec caused failures mentioned above.
